src/PythonQtPythonInclude.h

```diff
@@ -117,31 +117,23 @@
 #undef PYTHONQT_RESTORE_KEYWORDS
 #endif
 
-#if PY_MAJOR_VERSION >= 3
+#if PY_MAJOR_VERSION < 3
+#error "PythonQt requires Python >= 3.x"
+#endif
+
+// Optional compatibility shim for legacy wrappers generated by older PythonQt.
+// Enable by defining PYTHONQT_USE_PYSTRING_SHIM (deprecated).
+#if defined(PYTHONQT_USE_PYSTRING_SHIM)
 #define PY3K
-// Helper defines to facilitate porting
 #define PyString_FromString PyUnicode_FromString
```

Comment:
Maybe add a deprecation warning here? Or would this over-flood the log with warnings?